xmppserver/src/main/java/org/jivesoftware/openfire/roster/DefaultRosterItemProvider.java

```diff
@@ -58,26 +65,32 @@
             "INSERT INTO ofRoster (username, rosterID, jid, sub, ask, recv, nick) " +
             "VALUES (?, ?, ?, ?, ?, ?, ?)";
     private static final String UPDATE_ROSTER_ITEM =
-            "UPDATE ofRoster SET sub=?, ask=?, recv=?, nick=? WHERE rosterID=?";
+            "UPDATE ofRoster SET sub=?, ask=?, recv=?, nick=?, stanza=? WHERE rosterID=?";
     private static final String DELETE_ROSTER_ITEM_GROUPS =
             "DELETE FROM ofRosterGroups WHERE rosterID=?";
     private static final String CREATE_ROSTER_ITEM_GROUPS =
-            "INSERT INTO ofRosterGroups (rosterID, %s, groupName) VALUES (?, ?, ?)";
+            "INSERT INTO ofRosterGroups (rosterID, %s, groupName, stanza) VALUES (?, ?, ?, ?)";
     private static final String DELETE_ROSTER_ITEM =
             "DELETE FROM ofRoster WHERE rosterID=?";
     private static final String LOAD_USERNAMES =
             "SELECT DISTINCT username from ofRoster WHERE jid=?";
     private static final String COUNT_ROSTER_ITEMS =
             "SELECT COUNT(rosterID) FROM ofRoster WHERE username=?";
      private static final String LOAD_ROSTER =
-             "SELECT jid, rosterID, sub, ask, recv, nick FROM ofRoster WHERE username=?";
+             "SELECT jid, rosterID, sub, ask, recv, nick, stanza FROM ofRoster WHERE username=?";
     private static final String LOAD_ROSTER_ITEM_GROUPS =
              "SELECT ofRosterGroups.rosterID,groupName FROM ofRosterGroups " +
              "INNER JOIN ofRoster ON ofRosterGroups.rosterID = ofRoster.rosterID " +
              "WHERE username=? ORDER BY ofRosterGroups.rosterID, %s";
 
     private final Cache<String, LinkedList<RosterItem>> rosterItemCache = CacheFactory.createCache( "RosterItems" );
 
+    /**
```

Comment:
Is duplicating the pool of readers here defeating the purpose of the generic methods defined in ExternalizableUtil that was done in the previous commit?